test/app-luatest/http_client_test.lua

```diff
@@ -986,3 +986,36 @@ g.test_http_params_request_post = function(cg)
     t.assert_equals(resp.body, "k=2&v=3&v=4")
     t.assert_equals(resp.headers['content_type'], "application/x-www-form-urlencoded")
 end
+
```

Comment:
Regarding tests:

1. The motivating example is `&` or `=` in a key or a value. It would be nice to have such a case.
2. It would be nice to see a difference between `GET` and `POST`: have a space in the test case.
3. We have no test case that would show that a key name is encoded.

Reply:
> > 3. We have no test case that would show that a key name is encoded.
>
> Key contains a `&` now that is escaped.

Keys are `k` and `v` in this test and they don't contain `&`.